aces/Loris
# [Imaging] Sites/Project/subproject filters and permission

This PR:
1. modifies the way project/subproject/site filters are populated. Now those filters are displayed with values the current user has permission for.
2. fixed the data table for some modules (MRI violation), to remove entries for Projects and Sites the user does not have access to.
3. add some integration tests

Note:
The permission `imaging_browser_qc` was removed from [imaging_qc.class.inc](https://github.com/aces/Loris/blob/main/modules/imaging_qc/php/imaging_qc.class.inc#L189). This permission is not documented for this module and permission's description refers to the `imaging_browser` module. Can a senior developer confirm if documentation needs to be updated instead?

## To test
* Visit all the modules and check that all data entries belong to sites and projects the user has access to.
* Visit all the modules and check that the site/project/subproject menu filters contain entries the user has access to.
* Modified modules:
    - imaging_browser
    - imaging_qc
    - mri_violations
    - imaging_uploader

Resolves #6691
Related to #6829
Related to #6729
